### PR TITLE
add comma check to msftidy

### PR DESCRIPTION
**Pull request by firefart**
This adds a check for a extra comma to msftidy. This should adress stuff like

```
'Author'         =>
[
   'XXX <XXX[at]gmail.com>', #metasploit module
],
```

### CODE REVIEW

**Comment by wvu-r7** [comment]
This is the best thing ever.

File: tools/msftidy.rb [hunk 1]
```diff
@@ -601,6 +601,18 @@ def check_invalid_url_scheme
...
   end
 
+  # This checks looks for extranous commas
+  # Stuff with comments should match too: , #metasploit module ]
+  # but it should not match interpolation #{
+  def check_comma
+    test = @source.scan(/[^\n]+,(?:\s*#[^{][^\n]+)?\s*[\]\}]/)
+    unless test.empty?
+      test.each { |item|
+        info("Extra comma: #{item}")
```
Comment (wvu-r7) [inline tools/msftidy.rb]: Perhaps "extraneous comma" would be more appropriate, and it'd match your comment above. :)

File: tools/msftidy.rb [hunk 2]
```diff
@@ -601,6 +601,18 @@ def check_invalid_url_scheme
     end
   end
 
+  # This checks looks for extranous commas
+  # Stuff with comments should match too: , #metasploit module ]
+  # but it should not match interpolation #{
+  def check_comma
+    test = @source.scan(/[^\n]+,(?:\s*#[^{][^\n]+)?\s*[\]\}]/)
+    unless test.empty?
+      test.each { |item|
```
Comment (wvu-r7) [inline tools/msftidy.rb]: Since this is broken up into multiple lines, you might want to use ```do``` and ```end```.

Comment (wvu-r7) [inline tools/msftidy.rb]: Maybe make the same change on [line 598](https://github.com/rapid7/metasploit-framework/pull/4593/files#diff-ff5f6b3f4c780f2d2beb8c10726223b6R598).

**Comment by FireFart** [comment]
@wvu-r7 all changed. I think we need to fix some warnings before we can spot FP but i have not seen any so far

**Comment by jhart-r7** [comment]
Rubocop's current configuration will correctly flag (and fix, if you like) these situations, FWIW.

**Comment by wvu-r7** [comment]
@jhart-r7: We definitely need to make RuboCop more usable for MSF. ```msftidy``` isn't ideal to maintain, IMHO.

File: tools/msftidy.rb [hunk 3]
```diff
@@ -595,9 +595,21 @@ def check_udp_sock_get
...
-      test.each { |item|
+      test.each do |item|
         info("Invalid URL: #{item}")
-      }
+      end
+    end
+  end
+
+  # This checks looks for extranous commas
+  # Stuff with comments should match too: , #metasploit module ]
```
Comment (wvu-r7) [inline tools/msftidy.rb]: What's the bracket for?

Comment (FireFart) [inline tools/msftidy.rb]: `, #metasploit module ]` is the whole string that matches. Normaly it searches for `,]` and `,}` but there also comments between them.

Comment (wvu-r7) [inline tools/msftidy.rb]: Ah, nvm, I'm dumb. :)

**Pull request merged**